virtual_rainforest/models/animals/scaling_functions.py

```diff
@@ -31,23 +34,33 @@ def damuths_law(mass: float, terms: tuple) -> int:
     return ceil(terms[1] * mass ** terms[0])
 
 
-def metabolic_rate(mass: float, terms: tuple) -> float:
-    """The function to set the metabolic rate of animal cohorts.
-
-        Currently, this function provides the allometric scaling of the basal metabolic
-        rate of terrestrial mammals. This will be later expanded to be a more complex
-        function of metabolic type, functional type, activity levels, and temperature.
+def metabolic_rate(
+    mass: float, temperature: float, terms: tuple, metabolic_type: MetabolicType
+) -> float:
+    """Calculates the metabolic rate of animal cohorts.
 
     Args:
         mass: The body-mass [kg] of an AnimalCohort.
+        temperature: The temperature [Celsius] of the environment.
         terms: The tuple of metabolic rate terms used.
+        metabolic_type: The metabolic type of the animal [ENDOTHERMIC or ECTOTHERMIC].
 
     Returns:
         The metabolic rate of an individual of the given cohort in [J/s].
 
     """
-
-    return terms[1] * (mass * 1000) ** terms[0]
+    mass_g = mass * 1000  # Convert mass to grams
+    temperature_k = temperature + 273.15  # Convert temperature to Kelvin
+
+    if metabolic_type == MetabolicType.ENDOTHERMIC:
+        return terms[1] * mass_g ** terms[0]
+    elif metabolic_type == MetabolicType.ECTOTHERMIC:
+        b0, exponent = terms
+        return (
+            b0 * mass_g**exponent * exp(-0.65 / (BOLTZMANN_CONSTANT * temperature_k))
+        )
+    else:
+        raise ValueError("Invalid metabolic type: {metabolic_type}")
```

Comment:
Any reason why not using `match... case`? It was only a suggestion, but I'm wondering if it is just a choice or if you were having any technical problem with it. In principle, `mypy>=1.3` with Python 3.10 should work well with this.

```suggestion
    match metabolic_type:
        case MetabolicType.ENDOTHERMIC:
             return terms[1] * mass_g ** terms[0]
        case MetabolicType.ECTOTHERMIC:
             b0, exponent = terms
             return (
                 b0 * mass_g**exponent * exp(-0.65 / (BOLTZMANN_CONSTANT * temperature_k))
             )
```

Reply:
Oh, I tried it at first and then mypy got very angry about it so I shelved it for "later". The browsing I did online looked like it was a known problem but I may have misunderstood.

Reply:
It works now so it must have been something else I was doing! I'll add it in.

Reply:
I need to stop posting like things have been resolved before actually doing the push. 

Locally, flake8 and mypy accept the match/case structure. They get angry about it on github.

Reply:
Is that because the GitHub CI action uses python 3.9 for the CI build step?

Reply:
Yes. That is why 😄

We _could_ abandon support for `3.9` - there is then some other stuff that we could adopt, like the use of `None | str` in typing and the like. But I think we should do that in a systematic way as a Project to update the code base to 3.10. I'd stick with what works with 3.9 here?

Reply:
Ups. I didn't realised that we were officially supporting 3.9. Sorry for that. Well, 3.10 and 3.11 come with a lot of improvements and speedups - and this Autumn we will have 3.12. So it will not be a bad move to abandone something that was already "old" (1.5 years old) when the project started.

But I totally agree with @davidorme , I would stick with 3.9 compatible stuff for now and make the transition in an organised way.

Reply:
Alright, thanks folks!